PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkCommonRunParametersPanel.kt

```diff
@@ -68,28 +75,30 @@ class SparkCommonRunParametersPanel(private val myProject: Project, private val
                     fill = GridConstraints.FILL_HORIZONTAL
                 }
             }
-            row { c(mainClassPrompt);                     c(mainClassTextField) }
+            row { c(mainClassPrompt); c(mainClassTextField) }
+            row { c(); c(errorMessageLabel) }
         }
 
         formBuilder.buildPanel()
     }
 
-    open val component: JComponent
+    val component: JComponent
         get() = submissionPanel
 
     fun setMainClassName(mainClassName: String) {
         mainClassTextField.text = mainClassName;
     }
 
-    fun getMainClassName() : String {
+    fun getMainClassName(): String {
         return mainClassTextField.text
     }
 
     @Throws(ConfigurationException::class)
     fun validateInputs() {
-        // Check for command arguments invisible chars
-        if (mainClassTextField.text.isNullOrBlank()) {
-            throw ConfigurationException("Main Class Name should not be null")

```

Comment:
Have you tried to throw `RuntimeConfigurationWarningConfiguration`?

Reply:
It will also block the saving.